scarlet/frame.py

```diff
@@ -105,5 +108,171 @@ def get_sky_coord(self, pixel):
                     "Invalid number of wcs dimensions: {0}".format(self.wcs.naxis)
                 )
             return tuple(c.item() for c in coord)
-
         return tuple(pixel)
+
+    @staticmethod
+    def from_observations(observations, target_psf = None, target_wcs = None, obs_id = None, coverage = 'union'):
+        """Generates a frame from a set of observations.
+
+        By default, this method will generate a frame from a set of observations by indentifying the highest resolution
+        and the smallest PSF and use them to construct a common frome for all observations in the set.
+
+        Parameters
+        ----------
+        observations: array of `scarlet.Observation` objects
+            array that contains Observations to match onto a common frame
+        psfs: `scarlet.PSF` or its arguments
+            Target PSF to which oll observations are to be deconvolved.If set to None we use the smallest PSF across
+            all observations and channels.
+        wcs: TBD
+            World Coordinates of the target frame
+        obs_id:
+            index of the reference observation. If not set to None, the observation with the smallest PSF and smallest
+            pixel is used
+        coverage: `string`
+            Sets the frame to incorporate all the pixels covered by all the model ('union')
+            or sets the frame to incorporate only the pixels vovered by all the observations ('intersection').
+            Default is 'union'.
+        """
+        assert coverage in ['union', 'intersection']
+        # Array of pixel sizes for each observation
+        pix_tab = []
+        # Array of psf size for each psf of each observation
+        fat_psf_size = None
+        small_psf_size = None
+        channels = []
+        #Create frame channels and find smallest and largest psf
+        for c, obs in enumerate(observations):
+            #Concatenate all channels
+            channels = channels + obs.frame.channels
+            #concatenate all pixel sizes
+            pix_tab.append(interpolation.get_pixel_size(interpolation.get_affine(obs.frame.wcs)))
+            h_temp = interpolation.get_pixel_size(interpolation.get_affine(obs.frame.wcs))
+            # Looking for the sharpest and the fatest psf
+            for psf in obs.frame._psfs.image:
+                psf_size = get_psf_size(psf)*h_temp
+                if (fat_psf_size is None) or (psf_size > fat_psf_size):
```

Comment:
small detail, but since this is an `or` condition I would change the order, since `fat_psf_size` will only be true once whereas the second condition might be true multiple times.

Reply:
Gotcha!

Reply:
Nope, because if fat_psf_size is `None`, it does not work.